pkg/kn/plugin/manager.go

```diff
@@ -0,0 +1,515 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"text/template"
+
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/spf13/cobra"
+)
+
+// Allow plugins to register to this slice for inlining
+var InternalPlugins PluginList
+
+// Interface describing a plugin
```

Comment:
I think for the plugin purposes we should only extract this interface from the knative/client, maybe move it to `types.go` file and keep manager.go in tact in current repository. The interface is actually only common code for plugins to use.